packages/odyssey-react/src/components/Button/Button.test.tsx

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import Button, { ButtonVariants } from "./Button";
+
+const buttonLabel = "Button Label";
+
+const checkVariantClassName = (variantName: ButtonVariants, variantClassName: string) => {
+  const { getByTestId } = render(
```

Comment:
consider using `getByText` here instead. We may one day have to expose data attributes like this but I'm hopeful we can instead help developers write more robust tests that have less flake and that don't rely on these types of attrs by making assertions against a11y attrs or text content. It also helps keep our bundle size ever so slightly smaller :joy: Test ids can be our last and final option if we run out of other approaches.

Reply:
My assumption is that testids would be useful for e2e testing within admin. We take a very similar approach in [core.components](https://github.com/okta/okta-ui/blob/master/packages/core/components/src/components/button/button.tsx#L46) I'm happy to axe it for this PR, but we should figure out if this is something that we need to add going forward.

Reply:
That is a scenario I considered as well and we may one day *have* to add some amount of `data-testid` into our code base. However, I want to start by keeping our unit tests focused on using assertions that align closely with how a real user will interact with the UI which means less data-* attrs and more text or a11y based selectors!